File: code/java/src/main/java/visitor/Destroyer.java
package visitor;

public class Destroyer implements Visitor{
    @Override
    public void visitPainting(Painting painting) {
        System.out.println("Destroyer destroys the painting");
    }

    @Override
    public void visitSculpture(Sculpture sculpture) {
        System.out.println("Destroyer destroys the sculpture");
    }


}

File: code/typescript/src/visitor/Destroyer.ts
import { Visitor } from "./Visitor";
import { Painting } from "./Painting";
import { Sculpture } from "./Sculpture";

export class Destroyer implements Visitor {
    visitPainting(painting: Painting): void {
        console.log("Destroyer destroys the painting");
    }

    visitSculpture(sculpture: Sculpture): void {
        console.log("Destroyer destroys the sculpture");
    }
}
